calibration/step-5-unify_measurements.py

```diff
@@ -78,26 +96,59 @@ def unify_measurements(
 
         # iterate over all the measurment group and take the maximum for each tensor and its channel
         if scales:
-            for measurement_json in measurements_jsons:
+            for idx, measurement_json in enumerate(measurements_jsons):
+                # for experts of moe, append results in all measurements
+                if use_ep and is_moe_experts(node_name):
+                    if node_name not in moe_experts_data:
+                        moe_experts_data[node_name] = node_values
+                    else:
+                        prefix, local_expert_id = analyze_expert_name(node_name)
+                        new_node_name = ".".join((prefix, str(expert_num * idx + local_expert_id)))
+                        assert new_node_name not in moe_experts_data
+                        moe_experts_data[new_node_name] = measurement_json[node_name]
+                    continue
+
                 for i in range(0, len(max_inputs)):
                     max_inputs[i] = max(
                         measurement_json[node_name]["inputs"][i], max_inputs[i])
                 if max_outputs is not None:
-                    max_outputs = max(
-                        measurement_json[node_name]["outputs"], max_outputs)
+                    # for moe op, keep max of the first, retain rest from other measurements
+                    if use_ep and is_moe(node_name) and idx > 0:
+                        max_outputs[0] = max(
+                            measurement_json[node_name]["outputs"][0], max_outputs[0])
+                        max_outputs.extend(measurement_json[node_name]["outputs"][1:])
```

Comment:
For the EP, we are concatenating the results across the ranks. The order matters -- do we have logic to handle it?

Reply:
EP rank orders denpend on the EP assignment logical of vllm, we can not get more information about rank order only with the measurement results. And this script requires users to offer card group, I think we should merge EP based on user-provided order. Do you have any suggestion?

Reply:
> EP rank orders depend on the EP assignment logical of vllm

Agreed, once the number of ranks is given, the `expert_map` is fixed. We can only merge the EP based on the new the `expert_map`. There is no room for users to customize the merge logic, except by changing the number of ranks.

@xuechendi  please correct me if I am wrong.

Reply:
> There is no room for users to customize the merge logic, except by changing the number of ranks.

Users will pass --groups, it will impact the merge logic

Reply:
based on my observation, Expert Map is like below
Say if we have 16 experts distributed on 8 cards
[-1, -1, -1, -,1, 0, 1, -1, -1, -1, -,1, -1, -1, -1, -,1, -1 -1]
vs 4 cards
[-1, -1, -1, -,1, 0, 1, 2, 3, -1, -,1, -1, -1, -1, -,1, -1 -1]
Adjacent experts will be assigned to same card, so you might use same logic to unify.

Reply:
> based on my observation, Expert Map is like below Say if we have 16 experts distributed on 8 cards [-1, -1, -1, -,1, 0, 1, -1, -1, -1, -,1, -1, -1, -1, -,1, -1 -1] vs 4 cards [-1, -1, -1, -,1, 0, 1, 2, 3, -1, -,1, -1, -1, -1, -,1, -1 -1] Adjacent experts will be assigned to same card, so you might use same logic to unify.

Currently this PR does use this logic to unify EP